pkg/internal/server/grpc/get_metadata_delta_test.go

```diff
@@ -244,3 +248,389 @@ func TestValidateGetMetadataDeltaRequest(t *testing.T) {
+				if ga.GetName() == "snap-with-content-no-snaphandle" {
+					vsc.Status.SnapshotHandle = nil
+				}
+				return true, vsc, nil
+			},
+			expectError:   true,
+			expStatusCode: codes.Unavailable,
+			expStatusMsg:  fmt.Sprintf(msgUnavailableInvalidVolumeSnapshotContentStatusFmt, "snap-with-content-no-snaphandle"),
+		},
+		{
+			// VolumeSnapshotContent associated with source snapshot has unexpected driver
+			apiRequest: &api.GetMetadataDeltaRequest{
+				BaseSnapshotName:   "snap-with-invalid-driver",
+				TargetSnapshotName: "snap-2",
+				Namespace:          "test-ns",
+				SecurityToken:      "token",
+				StartingOffset:     0,
+				MaxResults:         256,
+			},
+			mockVolumeSnapshotContent: func(action clientgotesting.Action) (handled bool, ret apiruntime.Object, err error) {
+				ga := action.(clientgotesting.GetAction)
+				vsc := volumeSnapshotContent(ga.GetName(), ga.GetNamespace(), th.DriverName)
+				if ga.GetName() == "snap-with-invalid-driver" {
+					vsc.Spec.Driver = *stringPtr("driver-unexpected")
+				}
+				return true, vsc, nil
+			},
+			expectError:   true,
+			expStatusCode: codes.InvalidArgument,
+			expStatusMsg:  fmt.Sprintf(msgInvalidArgumentSnaphotDriverInvalidFmt, "snap-with-invalid-driver", th.DriverName),
+		},
+		{
+			// VolumeSnapshotContent associated with target snapshot has unexpected driver
+			apiRequest: &api.GetMetadataDeltaRequest{
+				BaseSnapshotName:   "snap-1",
+				TargetSnapshotName: "snap-with-invalid-driver",
+				Namespace:          "test-ns",
+				SecurityToken:      "token",
+				StartingOffset:     0,
+				MaxResults:         256,
+			},
+			mockVolumeSnapshotContent: func(action clientgotesting.Action) (handled bool, ret apiruntime.Object, err error) {
+				ga := action.(clientgotesting.GetAction)
+				vsc := volumeSnapshotContent(ga.GetName(), ga.GetNamespace(), th.DriverName)
+				if ga.GetName() == "snap-with-invalid-driver" {
+					vsc.Spec.Driver = *stringPtr("driver-unexpected")
+				}
+				return true, vsc, nil
+			},
+			expectError:   true,
+			expStatusCode: codes.InvalidArgument,
+			expStatusMsg:  fmt.Sprintf(msgInvalidArgumentSnaphotDriverInvalidFmt, "snap-with-invalid-driver", th.DriverName),
+		},
+	} {
+		if tc.mockVolumeSnapshot != nil {
+			th.fakeSnapshotClient.PrependReactor("get", "volumesnapshots", tc.mockVolumeSnapshot)
+		}
+		if tc.mockVolumeSnapshotContent != nil {
+			th.fakeSnapshotClient.PrependReactor("get", "volumesnapshotcontents", tc.mockVolumeSnapshotContent)
+		}
+		csiReq, err := grpcServer.convertToCSIGetMetadataDeltaRequest(context.TODO(), tc.apiRequest)
+		if tc.expectError {
+			assert.Error(t, err)
+			st, ok := status.FromError(err)
+			assert.True(t, ok)
+			assert.Equal(t, st.Code(), tc.expStatusCode)
+			assert.Equal(t, st.Message(), tc.expStatusMsg)
+			continue
+		}
+		assert.NoError(t, err)
+		validateCSIGetMetadataDeltaRequest(t, tc.expectedCSIRequest, csiReq)
+	}
```

Comment:
It would be nice if you could add a "name" to each test case and run the body of the `for` loop in a `t.Run` sub-test - it makes the test cases visible in the output.